### ISSUE TITLE
Video is written as uploaded but not available on tiktok

### ISSUE THREAD

**Issue by Flowtter** [wkaisertexas/tiktok-uploader#96]
Hi,

simple example that is not working on my end:
```python
from tiktok_uploader.upload import upload_video

upload_video(
    "output.mp4",
    description="first upload #cool",
    cookies="cookies.txt",
    browser="firefox",
    comment=True,
    stitch=False,
    duet=False,
    headless=True,
)
```

Without activating the headless mode, I also have the issue.\
Logs (`Video posted successfully`) and screen (`video has been uploaded`) seems to show that everything is working and that the video has been uploaded, but the video is not available on tiktok.

This is on a new fresh account.

**Comment by redrickh** [comment]
Try my fork

**Comment by romanrsr** [comment]
> Try my fork
I have tried your fork, but the video is not available

**Comment by redrickh** [comment]
Without any error log or something its kinda hard to help

**Comment by Flowtter** [comment]
Hi @redrickh,

Your fork didn't work either.
Stuck in `[16:31:58] Uploading video file`

Here is the error stacktrace I got after waiting a minute:
```
Message: 
Stacktrace:
RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8
WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:189:5
NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:507:5
dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:132:16
```

**Comment by redrickh** [comment]
edit config.toml and replace the upload_confirmation, process_confirmation line to:
```
 upload_confirmation = "//div[@title]"
 process_confirmation = "//div[contains(@class, 'cover-selector-image-container')]/img[contains(@class, 'cover-selector-image')]"

```

**Comment by Flowtter** [comment]
I'm using https://github.com/redrickh/tiktok-uploader/commit/c678a9f76c8758150812aff9c49b73d4567e8f49

**Comment by redrickh** [comment]
weird. let me check, it worked yesterday.

**Comment by redrickh** [comment]
Try this: process_confirmation = "//canvas[@id='mobile-preview-canvas']"

**Comment by unrapidint** [comment]
Same issue with the latest version.

**Comment by yuuval** [comment]
@redrickh i've made a logger of the program. Thats the main issue: Message: "no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//video\"}\n
It's an 404. I am using headless mode with chrome. For a picture of the logs, contact me

**Comment by redrickh** [comment]
Try my fork
https://github.com/redrickh/tiktok-uploader
I'm using it rn in headless mode and works perfectly!

**Comment by yuuval** [comment]
Thanks alot!

Am Fr., 2. Feb. 2024 um 23:37 Uhr schrieb Sándor ***@***.***>:

**Comment by sweldon** [comment]
This was happening to me too last night, but changing this in `config.toml` today fixed it for me (combination of @redrickh's suggestions above - thank you!):
```
	upload_confirmation = "//div[@title]"
	process_confirmation = "//canvas[@id='mobile-preview-canvas']"
```

**Comment by sweldon** [comment]
Strangely, it has stopped working again. Even when using @redrickh 's fork. Not sure what's going on... seems to be very inconsistent.

**Comment by wkaisertexas** [comment]
How many videos are you trying to upload at once? You get rate limited at really low numbers (5-10 videos per hour)

**Comment by sweldon** [comment]
Just one video, that is good to know though thank you. I just confirmed that I can upload the video manually, so I don't think im being rate limited. If i was being rate limited would the `post_confirmation` portion fail?

Basically I'm seeing this, but no video is posted
```
[20:59:17] Navigating to upload page
[20:59:19] Uploading video file
[20:59:35] Removing split window
[20:59:36] Split window not found or operation timed out
[20:59:36] Setting interactivity settings
[20:59:36] Setting description

[21:02:14] Clicking the post button
[21:02:15] Video posted successfully
```

**Comment by wkaisertexas** [comment]
Yeah, I think you can still be limited even if you can post the video manually. When I was making this, that was something that I noticed and found strange. 

I think TikTok likes the presense of auto-generated content because it can be engaging to viewers (copyrighted movies, tv shows, etc.). However, being naive and having no safeguards would make for a very easy lawsuit. So, they place modest restrictions on people so that they have plausible deniability. 

This may seem tin-foil-hatty, but TikTok will detect the presence of the bot on the signin page and not allow you to automatically sign in. The whole `cookies.txt` is a way to get around that limitation. 

TLDR; TikTok probably knows, but they probably do not care so long as they have plausible deniability.

**Comment by sweldon** [comment]
Yeah not sure. I just tried a single 1-vid upload today after not attempting for 16 hours and same result. Says successful yet no vid :( Thank you for your input though I appreciate it.

**Comment by wkaisertexas** [comment]
No, it is broken in general right now. I can not fix it. I am having some weird issues with selenium on my laptop right now but this was a general trend with development that I saw

**Comment by sweldon** [comment]
Roger that thanks the info and for working on it. I'll keep trying things as well.

**Comment by redrickh** [comment]
Its working 100% here. My script uploading videos to tiktok with it.

**Comment by sweldon** [comment]
Bizarre. How often are you running your script? You're running the main branch of this repo right no forks? How are you calling upload_video?

**Comment by redrickh** [comment]
every 6 hours on my vps. No, I'm using my fork.

**Comment by sweldon** [comment]
Just checked out your fork and called `upload_video` the same way, and still not working for me, super weird.
```
pip install git+ssh://git@github.com/redrickh/tiktok-uploader.git@main
...
Successfully built tiktok-uploader
Installing collected packages: tiktok-uploader
...
```
```
[14:38:01] Authenticating browser with cookies
[14:38:01] Create a chrome browser instance
[14:38:01] Authenticating browser with cookies
[14:38:03] Posting /home/steve/post_bot/output/Time Travel/Time Travel.mp4
               with description: Exploring the mind-bending concept of time travel!  #TimeTravel #CosmicJourney #Space
[14:38:03] Navigating to upload page
[14:38:05] Uploading video file
[14:38:26] Removing split window
[14:38:31] Split window not found or operation timed out
[14:38:31] Setting interactivity settings
[14:38:31] Setting description
[14:38:50] Clicking the post button
[14:38:50] Video posted successfully
```
... nothin on TikTok

**Comment by redrickh** [comment]
Maybe add your cookies again? Idk. The uploading process finishing sucessfully. Maybe try without headless Mode off and check whats going on.

**Comment by sweldon** [comment]
Yeah I have tried both fresh cookies and headless / regular mode. l am running it on Ubuntu WSL2 so no actual browser pops up, but this was working a few days ago so I fought that's the issue? Maybe I'll try moving it to windows so I can actually see the browser and test.

**Comment by redrickh** [comment]
yeah, you should try that. let me know if you figured it out.

**Comment by yuuval** [comment]
@redrickh it stopped working for me too sadly

**Comment by yuuval** [comment]
@redrickh Thats the only issue i've found
![image](https://github.com/wkaisertexas/tiktok-uploader/assets/91323166/745db51d-a65d-4dc1-94ad-518eb9732e88)

**Comment by redrickh** [comment]
> @redrickh Thats the only issue i've found ![image](https://private-user-images.githubusercontent.com/91323166/302104707-745db51d-a65d-4dc1-94ad-518eb9732e88.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcwNTQ4MzQsIm5iZiI6MTcwNzA1NDUzNCwicGF0aCI6Ii85MTMyMzE2Ni8zMDIxMDQ3MDctNzQ1ZGI1MWQtYTY1ZC00ZGMxLTk0YWQtNTE4ZWI5NzMyZTg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA0VDEzNDg1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUxODkzN2IyYjBiM2I5M2JkZTVkNTUzNmVhZmI1MGI0OWIwOTgzMzdkMzBlMTE1ZGNkYTEzOGY0ODIxOWRjODMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ciu13J6oQX8KLUML4SLgt3_Ofd31zeYh4luwv0uVusg)

With my fork?

**Comment by yuuval** [comment]
@redrickh yeah

**Comment by YoussefBechara** [comment]
> Yeah I have tried both fresh cookies and headless / regular mode. l am running it on Ubuntu WSL2 so no actual browser pops up, but this was working a few days ago so I doubt that's the issue? Maybe I'll try moving it to windows so I can actually see the browser and test.

broooo please i have THE EXACT SAME PROBLEM, pleaseee tell me when you solve it and how (i'll tell you too when i find out)

**Comment by infopcgood** [comment]
Same issue for me. I tried debugging the issue myself by making the program wait until it closes the browser. The run ends with this screen, with the post button clicked.
![20240219_02h22m04s_grim](https://github.com/wkaisertexas/tiktok-uploader/assets/112328126/4f257b39-39ce-4eb6-9702-598220a7c7e4)
However, when I go to 'Manage your posts' the video I posted is not there.
When I try to manually upload on the browser instance that Selenium has created, it still does not work. When I manually upload the video on my own browser, it works. I'm suspecting that TikTok is detecting and blocking this script.

**Comment by infopcgood** [comment]
> When I manually upload the video on my own browser, it works. I'm suspecting that TikTok is detecting and blocking this script.
Tried again, it doesn't work.
- Latest Firefox launched manually (with uBlock) : O
- Latest Chrome launched manually (clean) : X
- Latest Firefox launched by `browser="firefox"` : X
- Latest Chrome launched by script : X

**Comment by Paillat-dev** [comment]
Well @infopcgood do you have any updates?

**Comment by gnouv** [comment]
I think Tiktok knows which request is from bot, automatic browser so your submit will be not accepted.

use SeleniumWire, u can trace the XHR request when u post new video, Tiktok returns something like that:

{"log_pb":{"impr_id":"202402290...."},"status_code":0,"status_msg":"Sent successfully"}

It should be item_id (tiktok video ID) for valid upload, not text "Sent successfully"

**Comment by Paillat-dev** [comment]
jeyy @gnouv my fork currently works with chrome only as a browser if that can help https://github.com/Paillat-dev/tiktok-uploader

**Comment by Paillat-dev** [comment]
> > jeyy @gnouv my fork currently works with chrome only as a browser if that can help https://github.com/Paillat-dev/tiktok-uploader
> 
> same issue bro, no video uploaded.

A this point idk. I am running that from a dc with no proxy...
